### PR TITLE
🌐 Add Portuguese translation for `docs/pt/docs/advanced/templates.md`

### PR DESCRIPTION
**Pull request by BrunoFNRodrigues**
This PR translates the templates.md and add this page to the mkdocs.yml file.

### CODE REVIEW

File: docs/pt/docs/advanced/templates.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+```CSS hl_lines="4"
+{!../../../docs_src/templates/static/styles.css!}
+```
+
+E por você estar usando `StaticFiles`, esse arquivo CSS deveria ser servido automaticamente pela sua aplicação **FastAPI** na URL `/static/styles.css`.
+
+## Mais detalhes
+
+Para mais detalhes, incluindo como testar templates, cheque <a href="https://www.starlette.io/templates/" class="external-link" target="_blank">documentos do Starlette em templates</a>.
```
Comment (jvmazagao) [inline docs/pt/docs/advanced/templates.md:77]:
```suggestion
Para mais detalhes, incluindo como testar templates, cheque a <a href="https://www.starlette.io/templates/" class="external-link" target="_blank">documentação de templates do Starlette</a>.
```

File: docs/pt/docs/advanced/templates.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Templates
+
+Você pode usar qualquer template engine, que você quiser com o **FastAPI**.
```
Comment (jvmazagao) [inline docs/pt/docs/advanced/templates.md:3]:
```suggestion
Você pode usar qualquer *template engine*, que desejar com o **FastAPI**.
```

Comment (jvmazagao) [reply docs/pt/docs/advanced/templates.md:3]: I Highly suggest to use italic mode when using some English expression to use to be more readable.

Comment (jvmazagao) [reply docs/pt/docs/advanced/templates.md:3]: Another suggestion here is to join some sentences that in english have a lot of cohesion but when reading and speaking in portuguese sounds ambiguos, like `Voce pode utilizar qualquer *template engine* que desejar com **FastAPI**, como o Jinja2, o mesmo utilizado pelo Flask e outras ferramentas.`

**Review by jvmazagao — COMMENTED** [review]
Hello @BrunoFNRodrigues I made some suggestions for the PT Translation, can you check and analyze.

File: docs/pt/docs/advanced/templates.md (hunk 2 above)
Comment (BrunoFNRodrigues) [reply docs/pt/docs/advanced/templates.md]: Now that you mention it, i can see the ambiguity

**Review by jvmazagao — APPROVED** [review]
🚀

File: docs/pt/docs/advanced/templates.md (hunk 2 above)
Comment (PabloEmidio) [reply docs/pt/docs/advanced/templates.md]: Is there any reason why do not translate `template engine` as `gerador de templates`?

At the company a work to, we use this term all the time when handle template engine (like jinja2, as example).

I'd had the same experience talking to others developers as well

**Comment by LorhanSohaky** [comment]
@BrunoFNRodrigues, can you update this PR?

File: docs/pt/docs/advanced/templates.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Templates
+
+Você pode utilizar qualquer gerador de templates que desejar com **FastAPI**, como o Jinja2, o mesmo utilizado pelo Flask e outras ferramentas.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:3]:
```suggestion
Você pode utilizar qualquer gerador de templates que desejar com **FastAPI**.

Uma escolha comum é o Jinja2, o mesmo utilizado pelo Flask e outras ferramentas.
```

File: docs/pt/docs/advanced/templates.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+Existem recursos para configurar ele facilmente, que você pode usar diretamente na sua aplicação **FastAPI** (fornecido por Starlette).
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:5]:
```suggestion
Existem recursos para configurá-lo facilmente que você pode usar diretamente na sua aplicação **FastAPI** (fornecido por Starlette).
```

File: docs/pt/docs/advanced/templates.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+## Instalar dependências
+
+Instalar `jinja2`:
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:9]:
```suggestion
Instale `jinja2`:
```

File: docs/pt/docs/advanced/templates.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+</div>
+
+## Usando `Jinja2Templates`
+
+* Importe `Jinja2Templates`.
+* Crie um objeto `templates`, que você poderá reutilzar depois.
+* Declare um parâmetro `Request` na *operação de caminho* que irá retornar um template.
+* Use o `templates` que foi criou para dar `render` e retornar um `TemplateResponse`, passando o `request` como um par de valores chave no "contexto" do Jinja2.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:26]:
```suggestion
* Use os `templates` que  você criou para renderizar e retornar um `TemplateResponse`, passando o `request` como um dos pares chave-valor no "contexto" do Jinja2.
```

File: docs/pt/docs/advanced/templates.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+```Python hl_lines="4  11  15-16"
+{!../../../docs_src/templates/tutorial001.py!}
+```
+
+!!! note
+    Note que você tem que passar o `request` como parte do par de valores chave no contexto para o Jinja2. Então, você também tem que declarar ele na sua *operação de caminho*.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:33]:
```suggestion
    Note que você tem que passar o `request` como parte dos pares chave-valor no contexto para o Jinja2. Então, você também tem que declarar ele na sua *operação de caminho*.
```

File: docs/pt/docs/advanced/templates.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+!!! tip "Dica"
+    Ao declarar `response_class=HTMLResponse` o docs UI vai ser capaz de saber que a resposta vai ser em HTML.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:36]:
```suggestion
    Ao declarar `response_class=HTMLResponse` o docs UI vai ser capaz de saber que a resposta será em HTML.
```

File: docs/pt/docs/advanced/templates.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+```CSS hl_lines="4"
+{!../../../docs_src/templates/static/styles.css!}
+```
+
+E por você estar usando `StaticFiles`, esse arquivo CSS deveria ser servido automaticamente pela sua aplicação **FastAPI** na URL `/static/styles.css`.
+
+## Mais detalhes
+
+Para mais detalhes, incluindo como testar templates, cheque a<a href="https://www.starlette.io/templates/" class="external-link" target="_blank">documentação de templates do Starlette</a>.
```
Comment (LorhanSohaky) [inline docs/pt/docs/advanced/templates.md:75]:
```suggestion
Para mais detalhes, incluindo como testar templates, cheque a <a href="https://www.starlette.io/templates/" class="external-link" target="_blank">documentação de templates do Starlette</a>.
```

**Review by LorhanSohaky — COMMENTED** [review]
Acho que com essas mudanças a leitura ficará mais fluida

File: docs/pt/docs/advanced/templates.md (hunk 3 above)
Comment (PabloEmidio) [reply docs/pt/docs/advanced/templates.md:3]: Eu removeria o "que desejar", já está implícito no "você pode utilizar qualquer gerador de templates".
```suggestion
Você pode utilizar qualquer gerador de templates com **FastAPI**, como o Jinja2, o mesmo utilizado pelo Flask e outras ferramentas.
```

Comment (LorhanSohaky) [reply docs/pt/docs/advanced/templates.md:3]: Acho válido a sugestão

> Eu removeria o "que desejar", já está implícito no "você pode utilizar qualquer gerador de templates".

File: docs/pt/docs/advanced/templates.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+!!! note
+    Note que você tem que passar o `request` como parte do par de valores chave no contexto para o Jinja2. Então, você também tem que declarar ele na sua *operação de caminho*.
+
+!!! tip "Dica"
+    Ao declarar `response_class=HTMLResponse` o docs UI vai ser capaz de saber que a resposta vai ser em HTML.
+
+!!! note "Detalhes Técnicos"
+    Você também pode usar `from starlette.templating import Jinja2Templates`.
+
+    O **FastAPI** prove o mesmo `starlette.templating` como `fastapi.templating`, apenas para a sua conveniência, o(a) desenvolvedor(a). Porém, a maioria das respostas disponíveis vem diretamente do Starlette. O mesmo acontece com o `Request` e o `StaticFiles`.
```
Comment (PabloEmidio) [inline docs/pt/docs/advanced/templates.md:41]: Necessita de acentuação no verbo `provê`.

```suggestion
    O **FastAPI** provê o mesmo `starlette.templating` como `fastapi.templating`, apenas para a sua conveniência, o(a) desenvolvedor(a). Porém, a maioria das respostas disponíveis vem diretamente do Starlette. O mesmo acontece com o `Request` e o `StaticFiles`.
```

**Comment by tiangolo** [comment]
📝 Docs preview for commit df67cc4d3b403c14d10374744f10a886f5f1f214 at: https://649a1e777ba2251706e3aabb--fastapi.netlify.app

**Comment by codespearhead** [comment]
A new PR was created ( #11338 ) and have been recently merged.

**Pull request closed without merging**